Unable to deploy Azure marketplace managed application via terraform

Need help with identifying solution to the following error when instantiating a managed app on Azure marketplace via terraform. The managed application is currently in a private preview. 

resource  appears to end up being deployed in Azure but terraform either timesout after several minutes on trying to deploy the resource or it returns an error messages

```
Stack trace from the terraform-provider-azurerm_v2.67.0_x5 plugin:
panic: interface conversion: interface {} is float64, not string
goroutine 163 [running]:
github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/managedapplications.flattenManagedApplicationParametersOrOutputs(0x5664ae0, 0xc001bb2690, 0x10)
/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/managedapplications/managed_application_resource.go:369 +0x21e
github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/managedapplications.resourceManagedApplicationRead(0xc0006fd780, 0x5633920, 0xc0012b9500, 0x0, 0x0)
/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/managedapplications/managed_application_resource.go:250 +0xa85
github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/managedapplications.resourceManagedApplicationCreateUpdate(0xc0006fd780, 0x5633920, 0xc0012b9500, 0x0, 0x0)
/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/managedapplications/managed_application_resource.go:205 +0xbd0
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc000c8a8c0, 0x65892c8, 0xc0016382c0, 0xc0006fd780, 0x5633920, 0xc0012b9500, 0x0, 0x0, 0x0)
/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.go:318 +0x1ee
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc000c8a8c0, 0x65892c8, 0xc0016382c0, 0xc0002f3180, 0xc001549e00, 0x5633920, 0xc0012b9500, 0x0, 0x0, 0x0, ...)
/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.go:456 +0x67b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc0005a8d80, 0x65892c8, 0xc0016382c0, 0xc0009db630, 0xc0016382c0, 0x5afe1c0, 0xc000fd7900)
/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/grpc_provider.go:955 +0x8ef
github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).ApplyResourceChange(0xc0018af7a0, 0x6589370, 0xc0016382c0, 0xc0002f2f50, 0xc0018af7a0, 0xc000fd7980, 0xc001637ba0)
/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/server/server.go:332 +0xb5
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x5afe1c0, 0xc0018af7a0, 0x6589370, 0xc000fd7980, 0xc001b30c60, 0x0, 0x6589370, 0xc000fd7980, 0xc001642000, 0x673)
/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000230c40, 0x65d2fb8, 0xc000294a80, 0xc0015c3b00, 0xc000fd6ab0, 0x9ab7ba0, 0x0, 0x0, 0x0)
/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1292 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc000230c40, 0x65d2fb8, 0xc000294a80, 0xc0015c3b00, 0x0)
/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1617 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc000798c30, 0xc000230c40, 0x65d2fb8, 0xc000294a80, 0xc0015c3b00)
/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:940 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:938 +0x1fd
Error: The terraform-provider-azurerm_v2.67.0_x5 plugin crashed!
This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
╷
│ Error: Plugin did not respond
│ 
│ with azurerm_managed_application.sdmc,
│ on compute.tf line 1, in resource "azurerm_managed_application" "sdmc":
│ 1: resource "azurerm_managed_application" "sdmc" {
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵
```

--- a/azurerm/internal/services/managedapplications/managed_application_resource.go
+++ b/azurerm/internal/services/managedapplications/managed_application_resource.go
@@ -366,11 +366,13 @@ func flattenManagedApplicationParametersOrOutputs(input interface{}) map[string]
 
 	for k, v := range input.(map[string]interface{}) {
 		if v != nil {
-			switch _ := v.(type) {
+			switch t := v.(type) {
 			case float64:
 				results[k] = v.(map[string]interface{})["value"].(float64)
-			default:
+			case string:
 				results[k] = v.(map[string]interface{})["value"].(string)
+			default:
+				fmt.Errorf("unexpected parameter type %T", t)
 			}
 
 		}

--- a/azurerm/internal/services/managedapplications/managed_application_resource.go
+++ b/azurerm/internal/services/managedapplications/managed_application_resource.go
@@ -366,7 +366,13 @@ func flattenManagedApplicationParametersOrOutputs(input interface{}) map[string]
 
 	for k, v := range input.(map[string]interface{}) {
 		if v != nil {
-			results[k] = v.(map[string]interface{})["value"].(string)
+			switch _ := v.(type) {
+			case float64:
+				results[k] = v.(map[string]interface{})["value"].(float64)
+			default:
+				results[k] = v.(map[string]interface{})["value"].(string)
+			}
+
 		}
 	}
 

--- a/azurerm/internal/services/managedapplications/managed_application_resource.go
+++ b/azurerm/internal/services/managedapplications/managed_application_resource.go
@@ -247,11 +247,19 @@ func resourceManagedApplicationRead(d *pluginsdk.ResourceData, meta interface{})
 		}
 		d.Set("parameter_values", parameterValues)
 
-		if err = d.Set("parameters", flattenManagedApplicationParametersOrOutputs(props.Parameters)); err != nil {
+		parameters, err := flattenManagedApplicationParametersOrOutputs(props.Parameters)
+		if err != nil {
+			return err
+		}
+		if err = d.Set("parameters", parameters); err != nil {
 			return err
 		}
 
-		if err = d.Set("outputs", flattenManagedApplicationParametersOrOutputs(props.Outputs)); err != nil {
+		outputs, err := flattenManagedApplicationParametersOrOutputs(props.Outputs)
+		if err != nil {
+			return err
+		}
+		if err = d.Set("outputs", outputs); err != nil {
 			return err
 		}
 	}
@@ -358,10 +366,10 @@ func flattenManagedApplicationPlan(input *managedapplications.Plan) []interface{
 	return results
 }
 
-func flattenManagedApplicationParametersOrOutputs(input interface{}) map[string]interface{} {
+func flattenManagedApplicationParametersOrOutputs(input interface{}) (map[string]interface{}, error) {
 	results := make(map[string]interface{})
 	if input == nil {
-		return results
+		return results, nil
 	}
 
 	for k, v := range input.(map[string]interface{}) {
@@ -372,13 +380,13 @@ func flattenManagedApplicationParametersOrOutputs(input interface{}) map[string]
 			case string:
 				results[k] = v.(map[string]interface{})["value"].(string)
 			default:
-				fmt.Errorf("unexpected parameter type %T", t)
+				return nil, fmt.Errorf("unexpected parameter type %T", t)
 			}
 
 		}
 	}
 
-	return results
+	return results, nil
 }
 
 func flattenManagedApplicationParameterValuesValueToString(input interface{}) (string, error) {

